### PR TITLE
feat: Add support for Insert with row alias

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This adds planner support for new insert syntax
`insert into t(id, col) values(1, 'apa') as dt(x) on duplicate key col = x`

## Related Issue(s)

- Closes https://github.com/vitessio/vitess/issues/12122

## Checklist

-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/semantics/table_collector.go [hunk 1]
```diff
@@ -104,12 +109,98 @@ func (etc *earlyTableCollector) handleTableName(tbl sqlparser.TableName, aet *sq
 	etc.Tables = append(etc.Tables, tableInfo)
 }
 
+func (tc *tableCollector) visitRowAlias(ins *sqlparser.Insert, rowAlias *sqlparser.RowAlias) error {
```
Comment (systay) [inline go/vt/vtgate/semantics/table_collector.go:112]: nitpick: this method could do with a bit of refactoring love to make it easier to read. maybe a couple of `extract method` or something would be enough

**Review by systay — APPROVED** [review]

File: go/vt/vtgate/engine/insert.go [hunk 2]
```diff
@@ -363,6 +366,19 @@ func (ins *Insert) description() PrimitiveDescription {
...
+		var mids []string
+		for _, n := range ins.Mid {
+			mids = append(mids, sqlparser.String(n))
+		}
+		shardedQuery := ins.Prefix + strings.Join(mids, ", ") + ins.Alias + sqlparser.String(ins.Suffix)
+		if shardedQuery != ins.Query {
+			other["ActualQuery"] = shardedQuery
+		}
+	}
+
```
Comment (GuptaManan100) [inline go/vt/vtgate/engine/insert.go:381]: Was this code intended for debugging to see when the sharded query and the original query are different?

**Review by GuptaManan100 — APPROVED** [review]
Rest looks good to me!

File: go/vt/vtgate/engine/insert.go (hunk 2 above)
Comment (harshit-gangal) [reply go/vt/vtgate/engine/insert.go:381]: I kept this for a reason to fail in future if anything goes wrong in the final query output.

**Pull request merged**